File: Assessment/Assessment.Entities/QuestionService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class QuestionService
    {
        private IQuestionRepository _repository;

        public QuestionService(IQuestionRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Question question)
        {
            _repository.Insert(question);
        }

        public void Delete(int questionID)
        {
            _repository.Delete(questionID);
        }

        public void Update(Question question)
        {
            _repository.Update(question);
        }

        public Question Get(int questionID)
        {
            return _repository.Get(questionID);
        }

        public IEnumerable<Question> GetAll()
        {
            return _repository.GetAll();
        }

        public IEnumerable<Answer> GetAnswers(int questionID)
        {
            return _repository.GetAnswers(questionID);
        }
    }
}

File: Assessment/Assessment.WEB/src/app/services/question.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Question } from '../entities/question';

export class QuestionService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Question/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public getAnswers(id: number) {
        return this.http.get(this.baseUrl + this.controller + 'Answers/' + id);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(question: Question) {
        return this.http.put(this.baseUrl + this.controller, question);
    }

    public insert(question: Question) {
        return this.http.post(this.baseUrl + this.controller, question);
    }

    public delete(id: number) {
        return this.http.delete(this.baseUrl + this.controller + id);
    }
}